WalletWasabi/Crypto/ZeroKnowledge/ZkExponentProof.cs

```diff
@@ -0,0 +1,30 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public class ZkExponentProof
+	{
+		public ZkExponentProof(GroupElement publicPoint, GroupElement randomPoint, Scalar response)
+		{
+			Guard.False($"{nameof(publicPoint)}.{nameof(publicPoint.IsInfinity)}", publicPoint.IsInfinity);
+			Guard.False($"{nameof(randomPoint)}.{nameof(randomPoint.IsInfinity)}", randomPoint.IsInfinity);
+			Guard.False($"{nameof(response)}.{nameof(response.IsOverflow)}", response.IsOverflow);
+			if (publicPoint == randomPoint)
+			{
+				throw new InvalidOperationException($"{nameof(publicPoint)} and {nameof(randomPoint)} should not be equal.");
+			}
+
+			PublicPoint = publicPoint;
+			RandomPoint = randomPoint;
+			Response = response;
+		}
+
+		public GroupElement PublicPoint { get; }
```

Comment:
The public point should not be part of the proof, it should be known by the verifier in advance.

Reply:
According to @seresistvanandras it should.

Reply:
According to @seresistvanandras it's ok.